docs/ref/clvm.md

```diff
@@ -603,53 +603,224 @@ When used in an integer context, nil behaves as zero.
+| `sha256`            | 87        | 134          | 2             |
+| `ash`               | 596       | -            | 3             |
+| `lsh`               | 277       | -            | 3             |
+| `not`, `any`, `all` | 200       | 300          | -             |
+| `point_add`         | 101094    | 1343980      | -             |
+| `pubkey_for_exp`    | 1325730   | -            | 38            |
+|                     |           |              |               |
+| `CREATE_COIN`       | 1800000   | -            | -             |
+| `AGG_SIG_UNSAFE`,`AGG_SIG_ME` | 1200000 | -    | -             |
+
+Aside from cost, the maximum number of atoms or pairs in a CLVM program is 2^31. If this threshold is exceeded, the program will be terminated and no value will be returned.
+
+## Evaluating cost for a sample brun program
+
+In this section, we'll show you how to calculate the cost of a simple CLVM program by hand. The program we'll use is `brun (concat (q . fu) (q . bar))`.
+
+The `brun` command takes two arguments, a program and its "environment". If no environment is specified on the command line, we use an empty environment, "()".
+
+`brun` executes the program, passing in its arguments by producing a new program that looks like this:
+
+`(a 2 3)`, where:
+* `a` is the "apply" operator
+* `2` and `3` refer to the left and right halves of the binary tree that represents the whole CLVM program.
+
+At the lowest level of the interpreter, we interpret an atom as one of three things:
+
+1. A quote (cost 20)
+2. A path lookup into the environment (base cost of 44 + 4 for each bit, and some extra penalty if there are redundant leading zeros)
+3. An operator (mandatory cost of 1 + the cost of executing the operator)
+
+For the program `(a 2 3)`, the costs are as follows:
+* `a` eval (mandatory cost):   1
+* `a` (execution cost):       90
+* `2` (env lookup base cost): 44 
+* `2` (env lookup, one bit):   4
+* `3` (env lookup base cost): 44 
+* `3` (env lookup, one bit):   4
+
+Cost for `(a 2 3)` = 187
+
+Next we can calculate the cost of the original program, `(concat (q . fu) (q . bar))`:
+* `concat` eval (mandatory cost):   1
+* `q . fu` (cost of a quote):      20 
+* `q . bar` (cost of a quote):     20 
+* `concat` (execution cost):      142
+* `concat` arg cost ("fu"):       135 
+* `concat` arg cost ("bar"):      135
+* `concat` two bytes ("fu"):        6 (2 bytes * 3 cost per byte)
+* `concat` three bytes ("bar"):     9 (3 bytes * 3 cost per byte)
+* `malloc` five bytes ("fubar"):   50 (5 bytes * 10 malloc cost per byte)
+
+Cost for the original program = 518
+
+Gross cost = Cost for `(a 2 3)` + Cost for the original program = 187 + 518 = 705
+
+Finally, we must apply an offset to the cost. This involves subtracting the cost of the boiler-plate, as well as a constant offset of 53.
+
+Boiler-plate:
+* `a` execution cost:       90
+* `concat` execution cost: 142
+
+Boiler-plate cost = 232
```

Comment:
this isn't the boilerplate program. it's the cost of `(a 2 3)` and then 53 is subtracted.